### PR TITLE
feat(Topology.Algebra.Valuation): add integer_isOpen

### PR DESCRIPTION
**Pull request by mariainesdff**
The unit ball in a valued ring is open.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Valuation/Integers.lean [hunk 1]
```diff
@@ -36,6 +36,8 @@ def integer : Subring R where
   neg_mem' {x} hx := by simp only [Set.mem_setOf_eq] at hx; simpa only [Set.mem_setOf_eq, map_neg]
 #align valuation.integer Valuation.integer
 
+lemma mem_integer_iff (r : R) : r ∈ v.integer ↔  v r ≤ 1 := by rfl
```
Comment (alexjbest) [inline Mathlib/RingTheory/Valuation/Integers.lean:39]:
```suggestion
lemma mem_integer_iff (r : R) : r ∈ v.integer ↔ v r ≤ 1 := by rfl
```

File: Mathlib/Topology/Algebra/Valuation.lean [hunk 2]
```diff
@@ -166,4 +166,24 @@ theorem cauchy_iff {F : Filter R} : Cauchy F ↔
     exact h γ
 #align valued.cauchy_iff Valued.cauchy_iff
 
+variable (R)
+
+/-- The unit ball of a valued ring is open. -/
+theorem integer_isOpen :
+    IsOpen (_i.v.integer : Set R) := by
```
Comment (alexjbest) [inline Mathlib/Topology/Algebra/Valuation.lean:173]: Not that it matters but I'd be tempted to just put the statement on one line here

Comment (alexjbest) [reply Mathlib/Topology/Algebra/Valuation.lean:173]: Maybe we can rename the variable `_i` too, it looks very strange using that in the statement of a theorem

**Comment by digama0** [comment]
Note: I have pushed an update to the lean toolchain because this PR was on a [buggy version](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Potential.20data.20loss.20from.20.60lake.20clean.60.20with.204.2E2.2E0-rc2.2F3/near/397875701) of the toolchain. **WARNING**: checking out old commits of this PR using v4.2.0-rc2 or v4.2.0-rc3 can cause `lake clean` to delete your mathlib folder! If you need to do so, make sure to delete `lakefile.olean` manually before running any `lake` commands.

**Pull request closed without merging**